hbase-thrift/src/test/java/org/apache/hadoop/hbase/thrift/TestThriftServer.java

```diff
@@ -871,4 +878,32 @@ private void closeScanner(
     handler.scannerGet(scannerId);
     handler.scannerClose(scannerId);
   }
+
+  @Test
+  public void testGetThriftServerType() throws Exception {
+    ThriftHBaseServiceHandler handler =
+        new ThriftHBaseServiceHandler(UTIL.getConfiguration(),
+            UserProvider.instantiate(UTIL.getConfiguration()));
+    assertEquals(TThriftServerType.ONE, handler.getThriftServerType());
+  }
+
+  @Test
+  public void testGetThriftServerOneType() throws Exception {
+    // start a thrift2 server
+    HBaseThrift2TestingUtility THRIFT2_TEST_UTIL =
+        new HBaseThrift2TestingUtility();
+
+    LOG.info("Starting HBase Thrift Server Two");
+    THRIFT2_TEST_UTIL.startThriftServer(UTIL.getConfiguration());
+    try (TTransport transport = new TSocket(InetAddress.getLocalHost().getHostName(),
+        THRIFT2_TEST_UTIL.getServerPort())){
+      TProtocol protocol = new TBinaryProtocol(transport);
+      // This is our thrift client.
+      Hbase.Client client = new Hbase.Client(protocol);
+      // open the transport
+      transport.open();
+      assertEquals(TThriftServerType.TWO.name(), client.getThriftServerType().name());
+    }
+    THRIFT2_TEST_UTIL.stopThriftServer();
```

Comment:
wrap this in finally block? Otherwise its a thread leak if an exception is thrown in the try block above.

Reply:
make sense.